
Add DOWNLOAD macro (and needed qstrip).
This patch won't be useful anymore after the upgrade to Buildroot 2009.11.

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: buildroot/package/Makefile.autotools.in
===================================================================
--- buildroot.orig/package/Makefile.autotools.in	2009-10-23 16:09:52.000000000 +0200
+++ buildroot/package/Makefile.autotools.in	2009-10-23 16:10:11.000000000 +0200
@@ -117,6 +117,30 @@
 TERM_BOLD = #$(shell tput bold)
 TERM_RESET = #$(shell tput rmso)
 
+################################################################################
+# DOWNLOAD -- Download helper. Will try to download source from:
+# 1) BR2_PRIMARY_SITE if enabled
+# 2) Download site
+# 3) BR2_BACKUP_SITE if enabled
+#
+# Argument 1 is the source location
+# Argument 2 is the source filename
+#
+# E.G. use like this:
+# $(call DOWNLOAD,$(FOO_SITE),$(FOO_SOURCE))
+################################################################################
+
+# support make source-check/external-deps
+ifneq ($(SPIDER),)
+DOWNLOAD=$(WGET) -P $(DL_DIR) $(1)/$(2)
+else
+define DOWNLOAD
+	$(Q)test -e $(DL_DIR)/$(2) || \
+	for site in $(call qstrip,$(BR2_PRIMARY_SITE)) $(1) $(call qstrip,$(BR2_ALTERNATE_SITE)); \
+	do $(WGET) -P $(DL_DIR) $$site/$(2) && exit; done
+endef
+endif
+
 # Utility programs used to build packages
 TAR ?= tar
 #ACLOCAL_STAGING_DIR ?= $(STAGING_DIR)/usr/share/aclocal
Index: buildroot/Makefile
===================================================================
--- buildroot.orig/Makefile	2009-10-23 16:09:52.000000000 +0200
+++ buildroot/Makefile	2009-10-23 16:10:11.000000000 +0200
@@ -31,6 +31,9 @@
 	defconfig allyesconfig allnoconfig release tags \
 	source-check help
 
+# Strip quotes and then whitespaces
+qstrip=$(strip $(subst ",,$(1)))
+#"))
 
 # Use shell variables, if defined
 ifneq ($(BUILDROOT_LOCAL),)
